runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java

```diff
@@ -130,9 +134,13 @@ public void onGameStateChange(GameStateChanged event)
 				amount = 0;
 				break;
 			case LOGGED_IN:
-				if (config.amount() != -1 && !config.taskName().isEmpty())
+				if (client.getUsername() != lastUsername)
```

Comment:
This generally may work in this situation but you should not use ==/!= for string comparisons